src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1174,6 +1218,15 @@ void ThrowIfNotPeekLockMode()
             }
         }
 
+        void ThrowIfSessionLockLost()
+        {
+            if (this.LinkError != null &&
+                this.LinkError.Condition.Equals(AmqpClientConstants.SessionLockLostError))
+            {
```

Comment:
Not needed right? Anytime a link is closed and we are recreating a link, we will have the new trace that is describing the link error that was encountered on close anyway.